### PR TITLE
TC-ACE-1.4 automation

### PR DESCRIPTION
**Pull request by cecille**
Automation for test case https://github.com/CHIP-Specifications/chip-test-plans/pull/2364

Needs python rather than yaml so we can have wildcard reads and multiple endpoint returns

### CODE REVIEW

File: src/python_testing/TC_ACE_1_4.py [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+        asserts.assert_true(self.cluster is not None, "Invalid cluster name")
+        asserts.assert_true(self.attribute is not None, "Invalue attribute name")
+
+        self.print_step(1, "Commissioning, already done")
+
+        self.print_step(2, "TH1 writes ACL all clusters view on all endpoints")
+        admin_acl = Clusters.AccessControl.Structs.AccessControlEntry(privilege=Clusters.AccessControl.Enums.Privilege.kAdminister,
+                                                                      authMode=Clusters.AccessControl.Enums.AuthMode.kCase,
+                                                                      subjects=[],
+                                                                      targets=[Clusters.AccessControl.Structs.Target(endpoint=0, cluster=0x001f)])
```
Comment (tehampson) [inline src/python_testing/TC_ACE_1_4.py:108]: nit, can this magic cluster numbers be named constants? (here and all other spots of `cluster=` and `deviceType=`

Comment (cecille) [reply src/python_testing/TC_ACE_1_4.py]: I changed the ones for the clusters, but I'm not sure how one would access a list of device type IDs in python. Happy to make the change if you can point me at the location.

Note that we do have these hardcoded in the test case, so there is a reference for these, even if it's not in the script.

File: src/python_testing/TC_ACE_1_4.py [hunk 2]
```diff
@@ -0,0 +1,344 @@
...
+        asserts.assert_true(cluster in results[endpoint], err_msg)
+        err_msg = "Results not returned for attribute {} on ep {}".format(str(attribute), str(endpoint))
+        asserts.assert_true(attribute in results[endpoint][cluster], err_msg)
+        attr_ret = results[endpoint][cluster][attribute]
+        err_msg = "Error reading {}:{}".format(str(cluster), str(attribute))
+        asserts.assert_true(attr_ret is not None, err_msg)
+        asserts.assert_false(isinstance(attr_ret, Clusters.Attribute.ValueDecodeFailure), err_msg)
+
+    @async_test_body
+    async def test_TC_ACE_1_3(self):
```
[the remark this answers is not in the document]
Comment (cecille) [reply src/python_testing/TC_ACE_1_4.py]: 1.4. This is just the test name. I was writing the two of these at once, and must have missed changing.

File: src/python_testing/TC_ACE_1_4.py [hunk 3]
```diff
@@ -0,0 +1,344 @@
...
+    return getattr(cluster.Attributes, str)
+
+
+class TC_ACE_1_4(MatterBaseTest):
+
+    async def write_acl(self, acl):
+        # This returns an attribute status
+        result = await self.default_controller.WriteAttribute(self.dut_node_id, [(0, Clusters.AccessControl.Attributes.Acl(acl))])
+        asserts.assert_equal(result[0].Status, Status.Success, "ACL write failed")
+        print(result)
```
[the remark this answers is not in the document]
Comment (cecille) [reply src/python_testing/TC_ACE_1_4.py]: removed now.

File: src/python_testing/TC_ACE_1_4.py (hunk 1 above)
Comment (cecille) [reply src/python_testing/TC_ACE_1_4.py]: Spoke with Terence offline and the device type enums would need more code gen to create. Leaving this for now as this is an orthogonal problem.

File: src/python_testing/TC_ACE_1_4.py [hunk 4]
```diff
@@ -0,0 +1,344 @@
...
+import logging
+import sys
+
+import chip.clusters as Clusters
+import chip.clusters.Objects
+from chip.interaction_model import InteractionModelError, Status
+from matter_testing_support import MatterBaseTest, async_test_body, default_matter_test_main
+from mobly import asserts
+
+# This test requires several addition command line arguments
```
Comment (bzbarsky-apple) [inline src/python_testing/TC_ACE_1_4.py:27]:
```suggestion
# This test requires several additional command line arguments
```

**Review by andy31415 — APPROVED** [review]

**Review by krypton36 — APPROVED** [review]

**Review by jmeg-sfy — APPROVED** [review]
Q: why these Accesses tests cannot be done in Yaml ? to give me a better understanding thanks

**Review by woody-apple — APPROVED** [review]

**Pull request merged**